lib/linter.js

```diff
@@ -333,7 +333,8 @@ class Linter {
 
     for (let rule of rules) {
       try {
-        transform(templateAST, () => rule.getVisitor());
+        let visitor = await rule.getVisitor();
+        transform(templateAST, () => visitor);
```

Comment:
there is no `env` initially, and I removed it in other places,
getting 1 test error - 
![image](https://user-images.githubusercontent.com/1360552/104830242-06589100-588e-11eb-9710-79d510c11cf4.png)

Reply:
https://github.com/ember-template-lint/ember-template-recast/pull/458

Reply:
As far as I can tell, the only reason you might want `env` arg is to access `env.syntax`, but you can get that much easier from `require(‘ember-template-lint’).syntax` (or something like that)

Reply:
@rwjblue in general (after template recast changes), I think it’s a good idea, because in env we have code reference and module name, and we could resolve some async stuff, based on it

Reply:
But we don't need `env` to get either of those things, both are already on the plugin instance directly

Reply:
@rwjblue thank you! MR updated, now all green